### PR TITLE
Add eos_l2_interface module and integration tests

### PR DESCRIPTION
**Pull request by rcarrillocruz**
##### SUMMARY

Add eos_l2_interface module

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

eos_l2_interface

##### ANSIBLE VERSION

```
ansible 2.5.0 (add_eos_l2_interface c1c987cf44) last updated 2018/01/22 01:54:47 (GMT +200)
  config file = None
  configured module search path = [u'/home/ricky/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ricky/devel/ansible/lib/ansible
  executable location = /home/ricky/devel/ansible/bin/ansible
  python version = 2.7.14 (default, Nov  2 2017, 18:42:05) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]

```

### CODE REVIEW

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+DOCUMENTATION = """
+---
+module: eos_l2_interface
+version_added: "2.5"
+author: "Ricardo Carrillo Cruz (@rcarrillocruz)"
+short_description: Manage L@ interfaces on Arista EOS network devices.
```
Comment (ganeshrn) [inline lib/ansible/modules/network/eos/eos_l2_interface.py]: Typo here, should be L2 interfaces

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    obj = []
+
+    aggregate = module.params.get('aggregate')
+    if aggregate:
+        for item in aggregate:
+            for key in item:
+                if item.get(key) is None:
+                    item[key] = module.params[key]
+
+            item['name'] = item['name'].lower()
```
Comment (ganeshrn) [inline lib/ansible/modules/network/eos/eos_l2_interface.py:234]: Should this handle for space within interface name?
Something similar to  https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/eos/eos_vrf.py#L224 to avid idempotency related issues.

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            obj.append(item.copy())
+    else:
+        obj.append({
+            'name': module.params['name'].lower(),
```
Comment (ganeshrn) [inline lib/ansible/modules/network/eos/eos_l2_interface.py:238]: Same as above.

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    choices: ['access','trunk']
+  access_vlan:
+    description:
+      - Configure given VLAN in access port.
+        If C(mode=access), used as the access VLAN ID.
+  native_vlan:
+    description:
+      - Native VLAN to be configured in trunk port.
+        If C(mode=trunk), used as the trunk native VLAN ID.
+  trunk_allowed_vlans:
```
Comment (gundalow) [inline lib/ansible/modules/network/eos/eos_l2_interface.py:44]: Please add 
`alias: ['trunk_vlans']`

Bit confusing that the primary name is `trunk_allowed_vlans` though you use the aliased `trunk_vlans` in the examples.

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+    description:
+      - List of allowed VLANs in a given trunk port.
+        If C(mode=trunk), these are the only VLANs that will be
+        configured on the trunk, i.e. "2-10,15".
```
Comment (gundalow) [inline lib/ansible/modules/network/eos/eos_l2_interface.py]: `configured on the trunk, i.e. C(2-10,15).`

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+      - Configure given VLAN in access port.
+        If C(mode=access), used as the access VLAN ID.
+  native_vlan:
+    description:
+      - Native VLAN to be configured in trunk port.
+        If C(mode=trunk), used as the trunk native VLAN ID.
+  trunk_allowed_vlans:
+    description:
+      - List of allowed VLANs in a given trunk port.
+        If C(mode=trunk), these are the only VLANs that will be
```
Comment (gundalow) [inline lib/ansible/modules/network/eos/eos_l2_interface.py]: Perhaps change to `ONLY` to make that statement stand out a bit more

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+        configured on the trunk, i.e. "2-10,15".
+  aggregate:
```
Comment (gundalow) [inline lib/ansible/modules/network/eos/eos_l2_interface.py]: Please include an aggregate example

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+    access_vlan: 20
+
+- name: Ensure Ethernet1 only has vlans 5-10 as trunk vlans
+  eos_l2_interface:
+    name: Ethernet1
+    mode: trunk
+    native_vlan: 10
+    trunk_vlans: 5-10
+
+- name: Ensure Ethernet1 is a trunk port and ensure 2-50 are being tagged (doesn't mean others aren't also being tagged)
```
Comment (gundalow) [inline lib/ansible/modules/network/eos/eos_l2_interface.py]: We have `mode: trunk` here, so this appears to be the same as the above, and actually an ONLY example

File: lib/ansible/modules/network/eos/eos_l2_interface.py [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+    mode: trunk
+    native_vlan: 10
+    trunk_allowed_vlans: 2-50
+
+"""
+
+RETURN = """
+commands:
+  description: The list of configuration mode commands to send to the device
+  returned: always, except for the platforms that use Netconf transport to manage the device.
```
Comment (gundalow) [inline lib/ansible/modules/network/eos/eos_l2_interface.py]: What happens in the netconf case?

**Review by gundalow — COMMENTED** [review]
`ansible-test sanity --test pylint` failed with:

```
lib/ansible/modules/network/eos/eos_l2_interface.py:156:40: ansible-format-automatic-specification Format string contains automatic field numbering specification
lib/ansible/modules/network/eos/eos_l2_interface.py:163:44: ansible-format-automatic-specification Format string contains automatic field numbering specification
lib/ansible/modules/network/eos/eos_l2_interface.py:165:44: ansible-format-automatic-specification Format string contains automatic field numbering specification
```

File: lib/ansible/modules/network/eos/eos_l2_interface.py (hunk 1 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/eos/eos_l2_interface.py]: done

File: lib/ansible/modules/network/eos/eos_l2_interface.py (hunk 6 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/eos/eos_l2_interface.py]: done

File: lib/ansible/modules/network/eos/eos_l2_interface.py (hunk 4 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/eos/eos_l2_interface.py:44]: done

File: lib/ansible/modules/network/eos/eos_l2_interface.py (hunk 5 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/eos/eos_l2_interface.py]: done

File: lib/ansible/modules/network/eos/eos_l2_interface.py (hunk 9 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/eos/eos_l2_interface.py]: that's a copy pasta from nxos_l2_interface, there's no netconf on Arista. Removing from string...

File: lib/ansible/modules/network/eos/eos_l2_interface.py (hunk 2 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/eos/eos_l2_interface.py:234]: There's no interfaces param on this module.

**Review by gundalow — APPROVED** [review]

**Pull request merged**